### PR TITLE
feat(apps/frontend-*): add internationalization to student frontend

### PR DESCRIPTION
**Pull request by sjschlapbach**
This pull request adds English language support based on next-intl to the student frontend (frontend-pwa). As components in shared-components are partly also used by the lecturer frontends (frontend-manage and frontend-control), the corresponding library was already installed there as well.

### CODE REVIEW

File: apps/frontend-pwa/src/pages/editProfile.tsx [hunk 1]
```diff
@@ -44,32 +46,32 @@ const EditProfile: NextPageWithLayout = () => {
...
-            .min(5, 'Der Benutzername muss mindestens 5 Zeichen lang sein.')
-            .max(10, 'Der Benutzername darf nicht länger als 10 Zeichen sein.'),
+            .required(t('pwa.profile.usernameRequired'))
+            .min(5, t('pwa.profile.usernameMinLength', { length: '5' }))
+            .max(10, t('pwa.profile.usernameMaxLength', { length: '10' })),
           password: yup
             .string()
             .optional()
-            .min(8, 'Das Passwort muss mindestens 8 Zeichen lang sein.'),
+            .min(8, t('pwa.profile.passwordMinLenght', { length: '8' })),
```
Comment (rschlaefli) [inline apps/frontend-pwa/src/pages/editProfile.tsx]: length

File: apps/frontend-pwa/src/pages/editProfile.tsx [hunk 2]
```diff
@@ -44,32 +46,32 @@ const EditProfile: NextPageWithLayout = () => {
...
           passwordRepetition: yup.string().when('password', {
             is: (val: string) => val && val.length > 0,
             then: (schema) =>
               schema
-                .required('Passwörter müssen übereinstimmen.')
-                .min(8, 'Das Passwort muss mindestens 8 Zeichen lang sein.')
+                .required(t('pwa.profile.identicalPasswords'))
+                .min(8, t('pwa.profile.passwordMinLenght', { length: '8' }))
```
Comment (rschlaefli) [inline apps/frontend-pwa/src/pages/editProfile.tsx]: length

File: apps/frontend-pwa/src/pages/editProfile.tsx [hunk 3]
```diff
@@ -251,182 +255,32 @@ const EditProfile: NextPageWithLayout = () => {
...
-                          name="skinTone"
-                          style={{ width: '100%' }}
-                        >
-                          {AVATAR_OPTIONS.skinTone.map((value) => (
-                            <option key={value} value={value}>
-                              {AVATAR_LABELS[value]}
-                            </option>
-                          ))}
-                        </Field>
+                    {Object.keys(AVATAR_OPTIONS).map((key) => (
```
Comment (rschlaefli) [inline apps/frontend-pwa/src/pages/editProfile.tsx:264]: very nice simplification

**Review by rschlaefli — CHANGES_REQUESTED** [review]
Need to rework the approach for getStaticProps/getServerSideProps, and remove them from components (only relevant for pages).

File: apps/frontend-pwa/src/pages/editProfile.tsx (hunk 3 above)
Comment (sjschlapbach) [reply apps/frontend-pwa/src/pages/editProfile.tsx:264]: thanks :)

**Comment by sjschlapbach** [comment]
> Need to rework the approach for getStaticProps/getServerSideProps, and remove them from components (only relevant for pages).

Of course, this was a very non-ideal bug in the implementation... Thanks for pointing it out! Will be fixed right away.

File: apps/frontend-pwa/src/pages/editProfile.tsx (hunk 1 above)
Comment (sjschlapbach) [reply apps/frontend-pwa/src/pages/editProfile.tsx]: fixed

File: apps/frontend-pwa/src/pages/editProfile.tsx (hunk 2 above)
Comment (sjschlapbach) [reply apps/frontend-pwa/src/pages/editProfile.tsx]: fixed

**Review by rschlaefli — APPROVED** [review]

**Pull request merged**